### PR TITLE
feat: database schema update and beginnings of app cleanup

### LINKED ISSUE #20: feat: Local MySQL Database
### Describe the feature you'd like to request

Developers should be able to host and connect to their own MySQL database instance rather than the shared planetscale dev instance.

### Describe the solution you'd like to see

A local MySQL instance should be setup using docker compose and connected to using the mysql2 connector in drizzle. Developers should be able to switch between planetscale dev and their local db seamlessly.

### Additional information

_No response_

### LINKED ISSUE #25: feat: Design Relational Data Model
### Describe the feature you'd like to request

Moving into Milestone 2, we must have a well-thought-out design for our relational data model. The primary database is planetscale (MySQL). The data model should consider the "must have" requirements that we currently have planned.

### Describe the solution you'd like to see

An online whiteboard app (e.g., miro, lucidchart, figma) should be used to create an entity relationship diagram for the data model. This diagram should be shared within the team to be implemented in future sprints.

### Additional information

_No response_

### LINKED ISSUE #51: feat: Improve Code Quality after Milestone 1
### Describe the feature you'd like to request

PRs had to be rushed into main for the Milestone 1 presentation. After this milestone is delivered, it is important we retroactively improve the code quality to maintain a strong code foundation for future milestones.

### Describe the solution you'd like to see

Refactoring and testing the code written for Milestone 1, and improving the styling of the frontend.

### Additional information

_No response_

### LINKED ISSUE #56: bug: Failing Type Checks in CI and Commit Hooks
### Provide environment information

N/A

### Describe the bug

Currently, the typecheck step in our CI is failing. Typechecks are also failing locally in git commit hooks. These type issues should be fixed so that we can return to using these checks.

### Link to reproduction

https://github.com/restauwants/restauwants/actions/runs/7938304772/job/21676941991

### To reproduce

Run `pnpm run typecheck` on main and observe failures.

### Additional information

_No response_

### LINKED ISSUE #58: feat: Data Constraints on database
### Describe the feature you'd like to request

I should not be able to enter dates in the futures, negative dollar amounts, etc

### Describe the solution you'd like to see

Place data constraints on database entry form

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by callumcurtis**
Includes a new database schema and corresponding updates to the nextjs application. Also includes some code cleanup in the nextjs application after Milestone 1.

Includes #59 
Resolves #20 
Resolves #25 
Resolves #58 
Resolves #51 
Resolves #56

### CODE REVIEW

File: apps/nextjs/tailwind.config.ts
Comment (Nebula5102) [inline apps/nextjs/tailwind.config.ts:1]: It is fine that this was removed, but was the colour for the way the price text appeared altered in a different spot?

**Review by Nebula5102 — APPROVED** [review]
Good reformatting of code. Nice to see posts finally called by their correct name: reviews.

File: apps/nextjs/tailwind.config.ts
Comment (callumcurtis) [reply apps/nextjs/tailwind.config.ts:1]: Price is appearing in normal foreground color now - happy to change it back to green but I thought it was too bright. Custom colors should be defined in `globals.css` and then referred to in `tailwind.config.ts` so I removed the definitions from this file (approach from https://ui.shadcn.com/docs/theming#adding-new-colors).

Comment (Nebula5102) [reply apps/nextjs/tailwind.config.ts:1]: Okay, thank you. Everything looks good to me then. I will rebase and merge.

**Pull request merged**